services/src/main/java/org/exoplatform/news/rest/NewsTargetingRestResourcesV1.java

```diff
@@ -70,8 +72,9 @@ public Response getTargets(@Context HttpServletRequest request) {
   @ApiResponses(value = { @ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
       @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), })
   public Response getReferencedTargets(@Context HttpServletRequest request) {
+    org.exoplatform.services.security.Identity currentIdentity = ConversationState.getCurrent().getIdentity();
     try {
-      List<NewsTargetingEntity> referencedTargets = newsTargetingService.getReferencedTargets();
+      List<NewsTargetingEntity> referencedTargets = newsTargetingService.getReferencedTargets(currentIdentity);
       return Response.ok(referencedTargets).build();
     } catch (Exception e) {
```

Comment:
you need to catch IllegalAccessException and return Response.Status.UNAUTHORIZED, see NewsRestResourcesV1 methods